### PR TITLE
Min journal post page

### PR DESCRIPTION
**Pull request by MinieMoe**
## Describe your changes
- Implemented front end for home page where journal entries are posted:
  - Add bold fonts for post title
  - Made Post card component
  - Made Tag component
  - Used `Flatlist` as scroll page and implemented pull-to-refresh 
- Connected to Supabase to fetch posts:
  - Custom hook `useFetchJournalEntries` to fetch all posts, refresh for new posts, and whether still loading
  - Interfaces for Post data: `IJournalPost` and `IDBJournalPost`
 
## Issue ticket number and link
#54 

## Screenshots / Video
Add your media here...

## Checklist
- [X] I have performed a self-review of my code
- [X] I have used `npx expo-doctor` to review any dependency issues.
- [X] I have pulled and merged any recent changes from the main branch (no merge conflicts)
- [X] I have tested thoroughly any edge cases that might break the application
- [X] I have deleted any console.log statements and useless comments

### CODE REVIEW

**Review by dave-cao — CHANGES_REQUESTED** [review]
Nice work! 

Functionality questions
When we create a new text entry, and then go back to home, the new post won't show up until we refresh the page. Can we make it so it automatically shows up without needing to refresh?

Some styling changes / suggestions
- decrease the distance between posts a bit to fit the figma file
- add more shadow to bottom of post 

Figma:
image

Branch
image

File: components/user/home-screen/Post.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import {Image, StyleSheet, View, Text, TouchableOpacity} from 'react-native';
+import {Card} from '@rneui/themed';
+import {MonoText, TextSemiBold} from '@/components/StyledText';
+import Tag from './Tag';
+
+const CONTENT_LENGTH = 200;
+
+type PostProps = {
+  date: string; //TODO: or date object?
```
Comment (dave-cao) [inline components/user/home-screen/Post.tsx:10]: It's best practice to convert to date object, especially if we are going to use this more in the future.

**Review by dave-cao — CHANGES_REQUESTED** [review]
A few changes need to be made but great work so far!

**Comment by MinieMoe** [comment]
@dave-cao I updated the code based on your request

There are few things I to suggest in the next meeting:
- Update the name of `MonoText` in `StyledText.tsx` since it used "Poppins", not "Space Mono"
- Use interface `IJournalEntry` to enforce type whenever we need to use `JournalEntry` component or anything related to it (ex: create, fetch journal entry)

Other styling problem in the home page:
- There's a gap on top and bottom of the scrolling portion, I dont know what caused it or how to fix it

**Comment by MinieMoe** [comment]
@dave-cao I updated the code based on your request:
- reduce the spacing between each entry in the homepage
- add more shadow
- auto refresh for new entry on home page after submitting from text entry screen

There are few things I to suggest in the next meeting:
- Update the name of `MonoText` in `StyledText.tsx` since it used "Poppins", not "Space Mono"
- Use interface `IJournalEntry` to enforce type whenever we need to use `JournalEntry` component or anything related to it (ex: create, fetch journal entry)

Other styling problem in the home page:
- There's a gap on top and bottom of the scrolling portion, I dont know what caused it or how to fix it

**Review by dave-cao — APPROVED** [review]
Nice work!

**Pull request merged**